open_fdd/platform/drivers/onboard_driver.py

```
@@ -0,0 +1,164 @@
+            source="onboard",
+            metrics=["value"],
+            storage_ref=storage_ref,
+            success=True,
+            error="Synthetic onboard data ingested (OFDD_ONBOARD_ALLOW_SYNTHETIC enabled).",
+        )
+
+    def _request_json(path: str, *, method: str = "GET", payload: dict | None = None) -> list[dict]:
+        url = f"{base_url}{path}"
+        headers = {"X-OB-Api": api_key, "Content-Type": "application/json"}
+        data = None if payload is None else json.dumps(payload).encode("utf-8")
+        req = urllib.request.Request(url, method=method, headers=headers, data=data)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)